File: Greedy Algorithms/Smallest subset sum greater than all others.java
import java.util.*;

class Main{
    static int minElements(int[] arr){
        int totalSum=0;
        for(int x:arr) totalSum+=x;
        Arrays.sort(arr);
        int currSum=0,cnt=0;
        for(int i=arr.length-1;i>=0;i--){
            currSum+=arr[i];
            cnt++;
            if(currSum>totalSum-currSum)
                return cnt;
        }
        return cnt;
    }
    public static void main(String[] args){
        int[] arr={3,1,7,1};
        System.out.println(minElements(arr));
    }
}

File: Greedy Algorithms/Smallest subset sum greater than all others.cpp
#include<bits/stdc++.h>
using namespace std;

int minElements(vector<int>& arr){
    int totalSum=0;
    for(int x:arr) totalSum+=x;
    sort(arr.begin(),arr.end(),greater<int>());
    int currSum=0,cnt=0;
    for(int x:arr){
        currSum+=x;
        cnt++;
        if(currSum>totalSum-currSum)
            return cnt;
    }
    return cnt;
}
int main(){
    vector<int> arr={3,1,7,1};
    cout<<minElements(arr)<<endl;
    return 0;
}
